test/unit/rules/no-redundant-landmark-role-test.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const generateRuleTests = require('../../helpers/rule-test-harness');
+const rule = require('../../../lib/rules/no-redundant-landmark-role');
+
+const { createErrorMessage } = rule;
+
+generateRuleTests({
+  name: 'no-redundant-landmark-role',
+
+  config: true,
+
+  // TODO update with a good example that should pass
+  good: ['<form role="search"></form>', '<footer role="contentinfo"></footer>'],
```

Comment:
Can you add:

```hbs
<footer role={{this.foo}}></footer>
```

```hbs
<footer role="{{this.stuff}}{{this.foo}}"></footer>
```

Reply:
But I have a doubt, can we allow dynamic values for roles, the roles are not meant to be changed right? @MelSumner

Reply:
Mutation/rerendering is not the only reason you might want to use a path expression / concat statement. One example, is using API data to build up your page dynamically where you are looping over an array of results and using the data to completely control the rendered content. In that scenario it is completely unknown at _authoring time_ what the role should be, but instead it is controlled by the API data in some what (e.g. `role={{if (eq this.type 'article') 'main'}}`).

We absolutely must allow for `role={{something}}`.

Reply:
See https://github.com/ember-template-lint/ember-template-lint/issues/1286 for the type of error that I am trying to avoid. That **exact** error was caused by the assumption that `role` would always be a static value.